### PR TITLE
feat: add `set-time` endpoint

### PR DESCRIPTION
**Pull request by alex-Arc**
osc: `/ontime/set-time/<eventId>/[remaining,elapsed] <time in seconds>`
ws: `{"type": "set-time", "payload": {"eventId" : <eventId>, "property": "[remaining,elapsed]", "value": <time in seconds>}}`
image
image

notes:
https://github.com/cpvalente/ontime/issues/334#issuecomment-1877468901
https://github.com/cpvalente/ontime/issues/334#issuecomment-1877686988
#318 #115

### CODE REVIEW

File: packages/types/src/definitions/TimerType.type.ts [hunk 1]
```diff
@@ -3,4 +3,5 @@ export enum TimerType {
   CountUp = 'count-up',
   TimeToEnd = 'time-to-end',
   Clock = 'clock',
+  External = 'external',
```
Comment (cpvalente) [inline packages/types/src/definitions/TimerType.type.ts:6]: I am not sure about reusing this. 
The existing behaviour changes the way the time is presented.

I feel the external source is unrelated and I am not sure I would know to look for it here.

On the other hand, we can't provide a countdown countup to an event we don't have control over. 

Not really sure. What do you think?

File: apps/client/src/features/rundown/event-block/EventBlockInner.tsx [hunk 2]
```diff
@@ -10,6 +10,7 @@ import { IoPlayForward } from '@react-icons/all-files/io5/IoPlayForward';
 import { IoPlaySkipForward } from '@react-icons/all-files/io5/IoPlaySkipForward';
 import { IoStop } from '@react-icons/all-files/io5/IoStop';
 import { IoTime } from '@react-icons/all-files/io5/IoTime';
+import { MdInput } from '@react-icons/all-files/md/MdInput';
```
Comment (cpvalente) [inline apps/client/src/features/rundown/event-block/EventBlockInner.tsx:13]: I would prefer if we stick to Io5.

Either that or find a new icon set which has everything we need

File: apps/server/src/controllers/integrationController.ts [hunk 3]
```diff
@@ -267,6 +271,37 @@ export function dispatchFromAdapter(
       return { topic: 'timer', payload: timer };
     }
 
+    case 'set-time': {
+      const { eventId, property, value } = payload as ChangeOptions;
+      const amout = Number(value);
+      if (isNaN(amout)) {
+        throw new Error(`set-time "${value}" is not a number`);
+      }
+      const event = EventLoader.getEventWithId(eventId);
```
Comment (cpvalente) [inline apps/server/src/controllers/integrationController.ts:280]: Would we not like to fail faster?

Maybe add a method in the playback service to get current event and compare that?

File: apps/client/src/features/rundown/event-block/EventBlockInner.tsx (hunk 2 above)
Comment (alex-Arc) [reply apps/client/src/features/rundown/event-block/EventBlockInner.tsx:13]: agree, it is just a quick place holder

File: packages/types/src/definitions/TimerType.type.ts (hunk 1 above)
Comment (alex-Arc) [reply packages/types/src/definitions/TimerType.type.ts:6]: yes i'm also not sure

TimeToEnd and External can't really work at the same time
Clock and External doesn't make much sense either (I think?)

maybe both External Up and Down could be useful to some one 
although I expect that most people would end up using CountDown with external..?

I think maybe it is OK to say that External will always be a CountDown 
or we could add an ExternalUp and ExternalDown
(^maybe one of thees for v2)
or a separate toggle

and then we can revisit for v3,

Comment (cpvalente) [reply packages/types/src/definitions/TimerType.type.ts:6]: How about a single toggle "source: external" which when is set we disable the timer type in the interface?

File: apps/client/src/features/rundown/event-block/EventBlockInner.tsx (hunk 2 above)
Comment (alex-Arc) [reply apps/client/src/features/rundown/event-block/EventBlockInner.tsx:13]: IoFlash 
image

IoArrowForwardCircleOutline 
image

IoDiscOutline 
image

MdOutlineSettingsInputSvideo 
image

**Pull request closed without merging**